### PR TITLE
MCR-3114 support for dedicated tika server for indexing

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3114).

based on #2166

change base to main if #2166 is merged

### CODE REVIEW

File: mycore-solr/src/main/java/org/mycore/solr/index/file/tika/MCRSimpleTikaMapper.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+package org.mycore.solr.index.file.tika;
+
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import org.apache.solr.common.SolrInputDocument;
+import org.mycore.common.config.annotation.MCRProperty;
+
+import com.google.gson.JsonElement;
```
Comment (yagee-de) [inline mycore-solr/src/main/java/org/mycore/solr/index/file/tika/MCRSimpleTikaMapper.java]: Do not use Gson!

Comment (yagee-de) [reply mycore-solr/src/main/java/org/mycore/solr/index/file/tika/MCRSimpleTikaMapper.java]: We agreed to use Jackson only for new code and move away from GSON completely - somewhat later.

Comment (sebhofmann) [reply mycore-solr/src/main/java/org/mycore/solr/index/file/tika/MCRSimpleTikaMapper.java]: Changed to jackson. (which has worse api in this case)

**Review by toKrause — DISMISSED** [review]

File: mycore-solr/src/test/java/org/mycore/solr/index/file/tika/MCRSimpleTikaMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import junit.framework.TestCase;
+
+public class MCRSimpleTikaMapperTest extends TestCase {
+
+    public String testJson;
+    public static final String X_TIKA_PARSED_BY = "X-TIKA:Parsed-By";
+
+    public MCRSimpleTikaMapperTest(){
+        super();
+        try(InputStream is =MCRClassTools.getClassLoader().getResourceAsStream("tika_test.json")){
```
Comment (rsteph-de) [inline mycore-solr/src/test/java/org/mycore/solr/index/file/tika/MCRSimpleTikaMapperTest.java]: The code is not formatted?

File: mycore-solr/src/main/java/org/mycore/solr/index/file/tika/MCRTikaHttpClient.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public <T extends Throwable> void extractText(InputStream is, ThrowingConsumer<TreeNode, T> responseConsumer)
+        throws IOException, T {
+        HttpPut httpPut = new HttpPut(url + "tika/text");
+        httpPut.setHeader("Accept", "application/json");
+        httpPut.setEntity(new InputStreamEntity(is));
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            try (CloseableHttpResponse response = httpClient.execute(httpPut)) {
```
Comment (rsteph-de) [inline mycore-solr/src/main/java/org/mycore/solr/index/file/tika/MCRTikaHttpClient.java]: I would combine these 2 try () statements

**Review by rsteph-de — CHANGES_REQUESTED** [review]

**Review by rsteph-de — APPROVED** [review]
looks good for me

**Pull request merged**